test/generic/MPoly-test.jl

```diff
@@ -1727,3 +1714,115 @@ end
   @test is_zero(one(S))
   test_Ring_interface_recursive(S)
 end
+
+# -------------------------------------------------------
+
+# Coeff rings needed for the tests below
+ZeroRing,_ = residue_ring(ZZ,1);
+ZZmod720,_ = residue_ring(ZZ, 720);
+
+## MPoly over ZeroRing
+@testset "Nilpotent/unit for ZeroRing[x,y]" begin
+  P,(x,y) = polynomial_ring(ZeroRing, ["x", "y"]);
+  @test is_nilpotent(P(0))
+  @test is_nilpotent(P(1))
+  @test is_nilpotent(x)
+  @test is_nilpotent(-x)
+  @test is_nilpotent(x+y)
+  @test is_nilpotent(x-y)
+  @test is_nilpotent(x*y)
```

Comment:
Instead of adding these tests (which are very similar each time) into multiple places, it would be better to test them to the general ring interface test suite in `test/Rings-conformance-tests.jl` so that the tests get run for all our ring implementations (well, at least all that use the conformance tests).

Of course this may then reveal further issues, but so be it.

Reply:
Just to summarize our discussion: we should merge this as-is. Some of the tests can't be easily done in the general conformance tests.

However, we should add *some* tests to the general conformance tests. I'll open an issue for that.